Paladin namespace is scoped for being instanced, but still global when possible

secretrobotron opened the pull request:
1. Subsystems and Paladin core objects enclosed in instance of Paladin object and automatically initialized
2. Paladin namespace is overwritten when new Paladin() occurs
3. Tests updated to use paladin instead of Paladin (and no Game wrapper)
4. Subsystems altered to return an instantiable object, which is constructed when Paladin.subsystem.init occurs

secretrobotron committed:
1. Subsystems and Paladin core objects enclosed in instance of Paladin object and automatically initialized
2. Paladin namespace is overwritten when new Paladin() occurs
3. Tests updated to use paladin instead of Paladin (and no Game wrapper)
4. Subsystems altered to return an instantiable object, which is constructed when Paladin.subsystem.init occurs

modeswitch commented on src/paladin.js:
```diff
@@ -6,646 +6,654 @@ Array.prototype.remove = function(from, to) {
-};
-
-function SpatialComponent( position, rotation ) {
+var Paladin = window.Paladin = function ( options ) {
```
I don't think we want to do this. It makes it harder to keep instances separated. If the client code wishes to do this then that's fine, but we shouldn't do it for them.

secretrobotron replied:
What in particular? window.Paladin?

The "var Paladin" part is just for the scope in this file (convenience).

modeswitch replied:
Yes, window.Paladin.

modeswitch replied:
Oops, nm. Comment withdrawn.

modeswitch commented:
Since we want to allow multiple simultaneous instances of paladin, having any global state is going to be messy. So I think "still global when possible" is not something we want to do here.

secretrobotron commented:
I'm glad you agree with clearing out the global scope. I left it in for legacy (just in case), but editing the tests turned out to be tremendously simple without it.

Consider it done.

secretrobotron committed:
1. Fixed misplaced option.setup() in Paladin constructor
2. Fixed up sloppy var in tests.
3. Got rid of Paladin global scope variables/objects

secretrobotron commented:
Fixed up a few things and got rid of globally-scoped Paladin objects. You can do everything with a Paladin instance now (and, optionally, within its setup function).
